api/api.go

```diff
@@ -0,0 +1,149 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/gorilla/mux"
+	"gym_management_system/db"
+	customErr "gym_management_system/errors"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type Server struct {
+	listenAddr string
+	store      db.Storage
+	validator  *CustomValidator
+}
+
+type apiFunc func(http.ResponseWriter, *http.Request) error
+
+type Error struct {
+	Error string `json:"error"`
+}
+
+func NewServer(listenAddr string, store db.Storage, validator *CustomValidator) *Server {
+	return &Server{
+		listenAddr: listenAddr,
+		store:      store,
+		validator:  validator,
+	}
+}
+
+func (s *Server) Run() {
+	r := mux.NewRouter()
+
+	staticDir := "/static/"
+	r.PathPrefix(staticDir).Handler(http.StripPrefix(staticDir, http.FileServer(http.Dir("static"))))
+	r.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "static/index.html")
+	})
+
+	// Define the API router
+	apiRouter := r.PathPrefix("/api").Subrouter()
+
+	// Public Endpoints
+	apiRouter.HandleFunc("/signup", makeHTTPHandleFunc(s.handleCreateAccount)).Methods("POST")
+	apiRouter.HandleFunc("/signup", func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "static/signup.html")
+	}).Methods("GET")
+	apiRouter.HandleFunc("/login", makeHTTPHandleFunc(s.handleLogin)).Methods("POST")
+	apiRouter.HandleFunc("/login", func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "static/login.html")
+	}).Methods("GET")
+	apiRouter.HandleFunc("/logout", makeHTTPHandleFunc(s.handleLogout)).Methods("POST")
+	apiRouter.HandleFunc("/events", makeHTTPHandleFunc(s.handleGetEvents)).Methods("GET")
+	apiRouter.HandleFunc("/events/{eventId}/entries", makeHTTPHandleFunc(s.handleGetEventEntries)).Methods("GET")
+	apiRouter.HandleFunc("/memberships", makeHTTPHandleFunc(s.handleGetMemberships)).Methods("GET")
```

Comment:
Extract these line to some `func (s *Server) attachPublicEndpoints() {...}`. Same goes for the rest of the handlers.